tarxx.h

```diff
@@ -101,20 +104,102 @@ namespace tarxx {
+        {
+            unsigned long pos = 0;
+
+            // fill a new block with old and new data if enough data
+            // is available to fill a whole block
+            if ((stream_block_used_ + size) >= BLOCK_SIZE) {
+                block_t block;
+
+                // copy saved data
+                std::copy_n(stream_block_.data(), stream_block_used_, block.data());
+
+                // copy from new data
+                const auto copy_from_new_data = BLOCK_SIZE - stream_block_used_;
+                std::copy_n(data, copy_from_new_data, block.data() + stream_block_used_);
+
+                // write new complete block
+                write(block);
+
+                pos += copy_from_new_data;
+                size -= copy_from_new_data;
+                stream_block_used_ = 0;
+            }
+
+            // write new block as long as we have enough data to fill a whole block
+            while (size >= BLOCK_SIZE) {
+                block_t block;
+                std::copy_n(data + pos, BLOCK_SIZE, block.data());
+                pos += BLOCK_SIZE;
+                size -= BLOCK_SIZE;
+                write(block);
+            }
+
+            // store remaining data for next call or stream_file_complete
+            std::copy_n(data + pos, size, stream_block_.data() + stream_block_used_);
+            stream_block_used_ += size;
+        }
+
+        void stream_file_complete(const std::string& filename, __mode_t mode, __uid_t uid, __gid_t gid, __off_t size, __time_t mod_time)
+        {
+            // create last block and fill with 0 if necessary
+            block_t block;
+            if (stream_block_used_ < block.size())
+                std::fill_n(block.begin() + stream_block_used_, block.size() - stream_block_used_, 0);
+            std::copy_n(stream_block_.data(), block.size(), block.data());
+            stream_block_used_ = 0;
+            write(block);
+
+            const auto stream_pos = file_.tellp();
+
+            // seek to header
+            file_.seekp(stream_file_header_pos_);
+            write_header(filename, mode, uid, gid, size, mod_time);
+            stream_file_header_pos_ = -1;
+
+            file_.seekp(stream_pos);
+        }
+
     private:
 
+        static constexpr const unsigned int HEADER_POS_MODE  = 100U;
+        static constexpr const unsigned int HEADER_POS_UID =  108U;
+        static constexpr const unsigned int HEADER_POS_GID =  116U;
+        static constexpr const unsigned int HEADER_POS_SIZE =  124U;
+        static constexpr const unsigned int HEADER_POS_MTIM = 136U;
+        static constexpr const unsigned int HEADER_LEN_MODE = 7U;
+        static constexpr const unsigned int HEADER_LEN_UID = 7U;
+        static constexpr const unsigned int HEADER_LEN_GID = 7U;
+        static constexpr const unsigned int HEADER_LEN_SIZE = 11U;
+        static constexpr const unsigned int HEADER_LEN_MTIM = 11U;
```

Comment:
`constexpr` implies `const`, please remove `const`
```suggestion
        static constexpr unsigned int HEADER_POS_MODE  = 100U;
        static constexpr unsigned int HEADER_POS_UID =  108U;
        static constexpr unsigned int HEADER_POS_GID =  116U;
        static constexpr unsigned int HEADER_POS_SIZE =  124U;
        static constexpr unsigned int HEADER_POS_MTIM = 136U;
        static constexpr unsigned int HEADER_LEN_MODE = 7U;
        static constexpr unsigned int HEADER_LEN_UID = 7U;
        static constexpr unsigned int HEADER_LEN_GID = 7U;
        static constexpr unsigned int HEADER_LEN_SIZE = 11U;
        static constexpr unsigned int HEADER_LEN_MTIM = 11U;
```